### ISSUE TITLE
Clean up + unify string concatenation

### ISSUE THREAD

**Issue by BasThomas** [rnystrom/GitHawk#1121]
We are doing (often similar or the same) string concatenation in multiple places, like adding the owner+repo name to create a URL.

I noticed this in #1120 (specifically, check [this](https://github.com/rnystrom/GitHawk/pull/1120#issuecomment-348324698) comment). We should look at how we can clean this up and prevent duplication that can break things in the future.

See [here](https://github.com/rnystrom/GitHawk/blob/a313f6e110e41bac83e38edcad28df637654bfcf/Classes/Models/CommitDetails.swift#L13-L14), [here](https://github.com/rnystrom/GitHawk/blob/e54b13b9e1c3a774afe4283a92103c5d0df9b6a1/Classes/Issues/IssueDetailsModel.swift#L13-L14), [here](https://github.com/rnystrom/GitHawk/blob/a504a87033662e7dc537793d4b61722fbd605435/Classes/Models/LabelDetails.swift#L13-L14), [here](https://github.com/rnystrom/GitHawk/blob/63cc844b0ab3ead3383de421931577dd14700c6d/Classes/Repository/RepositoryDetails.swift#L12-L13), [here](https://github.com/rnystrom/GitHawk/blob/3ac7e50293dcc25dfe3f8c5facbff0f6ce859e87/Classes/Issues/Referenced/IssueReferencedModel.swift#L21-L22), [here](https://github.com/rnystrom/GitHawk/blob/eb0e887f09e5877e5e54269b02a7edbbb966b401/Classes/Bookmark/BookmarkModel.swift#L23-L24), [here](https://github.com/rnystrom/GitHawk/blob/63cc844b0ab3ead3383de421931577dd14700c6d/Classes/Search/SearchRepoResult.swift#L20-L21), [here](https://github.com/rnystrom/GitHawk/blob/1e2e0c16985340a4e1f2d794574efaf8d5944be3/Classes/Bookmark%20v2/Bookmark.swift#L13-L14)... (and some other places - you get the idea).

And that excludes the actual concatenation that happens in probably as many places? 😅

**Comment by Sherlouk** [comment]
I know on RepositoryDetails I had a nice extension which did the concatenation so it was only done in one place, but this was removed -- can't remember why though

**Comment by BasThomas** [comment]
Yeah so I wanted to do the same when tackling #1120, but when I saw the issue was actually much bigger / more widespread, I held off on doing so. I think if we introduce a model encapsulating this, we should definitely add that as an extension / computed var!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>
